fbw-a380x/src/systems/systems-host/systems/FlightWarningSystem/FwsFlightPhases.ts

```diff
@@ -0,0 +1,519 @@
+    this.prevPhase11 = false;
+
+    this.groundImmediateMemo = new NXLogicTriggeredMonostableNode(2);
+    this.phase6Memo = new NXLogicTriggeredMonostableNode(15);
+    this.phase7Memo = new NXLogicTriggeredMonostableNode(120);
+    this.phase89Memo = new NXLogicTriggeredMonostableNode(180);
+
+    this.memoTo_conf01 = new NXLogicConfirmNode(120, true);
+    this.memoTo_memo = new NXLogicMemoryNode(false);
+
+    this.memoLdgMemo_conf01 = new NXLogicConfirmNode(1, true);
+    this.memoLdgMemo_inhibit = new NXLogicMemoryNode(false);
+    this.memoLdgMemo_conf02 = new NXLogicConfirmNode(10, true);
+    this.memoLdgMemo_below2000ft = new NXLogicMemoryNode(true);
+
+    this.memoToInhibit_conf01 = new NXLogicConfirmNode(3, true);
+
+    this.memoLdgInhibit_conf01 = new NXLogicConfirmNode(3, true);
+
+    // altitude warning
+    this.previousTargetAltitude = NaN;
+    this._wasBellowThreshold = false;
+    this._wasAboveThreshold = false;
+    this._wasInRange = false;
+    this._wasReach200ft = false;
+  }
+
+  update(deltaTime: number) {
+    this._updateFlightPhase(deltaTime);
+    this._updateTakeoffMemo(deltaTime);
+    this._updateLandingMemo(deltaTime);
+    this._updateAltitudeWarning();
+  }
+
+  _updateFlightPhase(_deltaTime: number) {
+    const raHeight1InValid = this.fws.radioHeight1.isFailureWarning() || this.fws.radioHeight1.isNoComputedData();
+    const raHeight2InValid = this.fws.radioHeight2.isFailureWarning() || this.fws.radioHeight2.isNoComputedData();
+    let radioHeight;
+    if (raHeight1InValid) {
+      if (raHeight2InValid) {
+        radioHeight = this.fws.radioHeight3;
+      } else {
+        radioHeight = this.fws.radioHeight2;
+      }
+    } else {
+      radioHeight = this.fws.radioHeight1;
+    }
+    // TODO find a better source for the following value ("core speed at or above idle")
+    // Note that N1 starts below idle on spawn on the runway, so this should be below 16 to not jump back to phase 1
+    const oneEngRunning =
+      this.fws.N1Eng1.get() > 15 ||
+      this.fws.N1Eng2.get() > 15 ||
+      this.fws.N1Eng3.get() > 15 ||
+      this.fws.N1Eng4.get() > 15;
+    const oneEngineRunning = this.oneEngineRunningConf.write(oneEngRunning, _deltaTime);
+    const noEngineRunning = !oneEngineRunning;
+    const hFail =
+      this.fws.radioHeight1.isFailureWarning() &&
+      this.fws.radioHeight2.isFailureWarning() &&
+      this.fws.radioHeight3.isFailureWarning();
+    const adcTestInhib = false;
+
+    const groundImmediate = Simplane.getIsGrounded();
+    const ground = this.gndMemo.write(groundImmediate, _deltaTime);
+
+    const ias = this.fws.computedAirSpeedToNearest2.get();
+    const acSpeedAbove80kts = this.speedAbove80KtsMemo.write(ias > 83, ias < 77);
+
+    const v1 = SimVar.GetSimVarValue('L:AIRLINER_V1_SPEED', 'knots');
+    let acAboveV1: boolean;
+    if (v1) {
+      acAboveV1 = this.speedAboveV1Memo.write(ias > v1 + 3, ias < v1 - 3);
+    } else {
+      acAboveV1 = false;
+    }
```

Comment:
```suggestion
    acAboveV1 = v1 ? this.speedAboveV1Memo.write(ias > v1 + 3, ias < v1 - 3) : false;
```

Reply:
while this improves the code, I'd not touch it right now. It's a 1:1 copy from the old code, and worked there